### DISCUSSION TITLE
Custom CSS Cascade Layers & Separate CSS Files Support

### DISCUSSION THREAD

**Discussion by koba04** [chakra-ui/panda#3038] [Ideas]
Hi, team👋 Thanks for all your great work!!!👏 

Panda uses CSS Cascade Layers and we put all layers like

```css
@layer reset, base, tokens, recipes, utilities;
```

That totally makes sense, but I want to customize layers (`@foundation`, `@component`) exposing to an application as the following not to expose internal layers (`reset`, `base`, `tokens`, `recipes`, `utilities`) PandaCSS utilizes.

```css
/* design-system.css */

/* This layer has styles for the global scope */
@layer design-system-foundation {
  @layer reset, base;
}

@layer design-system-component {
  @layer tokens, recipes, utilities;
}
```

And then, the application can use those layers like

```css
@layer design-system-foundation, application-foundation design-system-component, application; 
```

The application doesn't have to know the internal layer structure and it brings us an abstraction that is useful to organize CSS architecture for the application.

ref. https://css-tricks.com/css-cascade-layers/#aa-managing-a-complex-css-architecture

This also enables us to have separate CSS files for specific layers.

```css
/* foundation.css */
@layer reset, base;
```

Unfortunately, this doesn't work because Panda checks if `@layer` has all layers

https://github.com/chakra-ui/panda/blob/main/packages/core/src/context.ts#L358-L361

So I'd like to suggest to be able to define each CSS Cascade Layers like

- Nested Layers Support

```css
/* design-system.css */

/* This layer has styles for the global scope */
@layer design-system-foundation {
  @layer reset, base;
}

@layer design-system-component {
  @layer tokens, recipes, utilities;
}
```

- Separate CSS Files Support

```css
/* foundation.css */
@layer reset, base;
```

```css
/* component.css */
@layer tokens, recipes, utilities;
```

Does this make sense?

**Comment by segunadebayo** [comment]
You can achieve this today without any Panda changes. Just nest Panda's layer declaration inside your parent layer:

```css
@layer design-system, app;

@layer design-system {
  @layer reset, base, tokens, recipes, utilities;
  /* Panda styles go here */
}

@layer app {
  /* your app styles here */
}
```

Panda's internal layers work correctly within the nested scope. Your app layer takes priority. Consumers don't need to know about the internal layer structure.
